File: MJ/28250.py
# https://www.acmicpc.net/problem/28250
# 이브, 프시케 그리고 푸른 MEX의 아내

import sys
n = int(sys.stdin.readline())
arr = list(map(int, sys.stdin.readline().split(" ")))
count_zero, count_one, count_other = 0,0,0

for i in range(n):
    if arr[i]==0: count_zero+=1
    elif arr[i]==1: count_one+=1
    else: count_other+=1
print(int(count_zero*(count_zero-1)/2+count_one*count_zero*2+count_other*count_zero))

File: MJ/28250.cpp
#include <iostream>
using namespace std;

int main() {
ios_base :: sync_with_stdio(false); 
cin.tie(NULL); 
cout.tie(NULL);
//////////////////////////////////////////////////
int n; cin >> n;
long long countZero = 0, countOne = 0, countOther = 0;
for (int i=0;i<n;i++){
    int num; cin >> num;
    if (num==0) countZero++;
    else if (num==1) countOne++;
    else countOther++;        
}
long long ans =countOne*countZero*2 + countOther*countZero + (countZero*(countZero-1))/2;
cout << ans;
//////////////////////////////////////////////////
return 0;}